docs/deprecated-sql-function.md

```diff
@@ -0,0 +1,195 @@
+
+## Deprecated
+As its name states, it is deprecated, that means, it will be removed in future versions.
+
+## Trade-offs
+Performance and stability might get compromised when using this analysis depending on the function. With great power comes great responsibility.
+
+## Function requirements
+The word _arbitrary_ gets us into the unrestricted territory. However, functions compatible with the `deprecated-sql-function` must follow some conventions and adhere to some rules.
+
+ - Naming convention: funtions names must start with `DEP_EXT_`.
+ - Funtions must implement two operations:
+   1. The creation of a table to store the results.
+   2. The population of that very same table.
+ - Funtions must handle, at least, the following signature:
+   * First parameter: the operation to perform. Either `create` or `populate` the table.
+   * Second parameter: the name of the table to create or populate.
+   * Third parameter: a _free query_ from the primary source of the analysis.
+   * Forth parameter: an array of `text` listing all the columns from the _free query_ in the third parameter.
+   * Third, and consecutive parameters: it can receive an arbitrary number of `string`s and `number`s as extra parameters.
+ - Functions might handle an optional signature where first, second, third and forth are the same as in the previous signature, but:
+   * Fifth parameter: a _free query_ from the secondary source of the analysis.
+   * Sixth parameter: an array of `text` listing all the columns from the _free query_ in the fifth parameter.
+   * Seventh, and consecutive parameters up to second-to-last: it can receive an arbitrary number of `string`s and `number`s as extra parameters.
+ - The supported parameters types are: numeric, text, and boolean.
+
+## Examples
+
+Let's create a couple of examples based on existing analyses.
+
+### Buffer
+
+In the `buffer` analysis we want to include all existing columns but we want to expand the `the_geom` column with a
+buffer in determined by an in meters `radius` param.
+
+```sql
+CREATE OR REPLACE FUNCTION DEP_EXT_buffer(
+        operation text, table_name text, primary_source_query text, primary_source_columns text[], radius numeric
+    )
+    RETURNS VOID AS $$
+        DECLARE selected_columns TEXT;
+        BEGIN
+            -- We want to drop original `the_geom` column as we are gonna return a different one.
+            SELECT array_to_string(
+              array_cat(
+                array_remove(primary_source_columns, 'the_geom'),
+                -- And we add our new `the_geom` column using the `radius` argument.
+                ARRAY['ST_Buffer(the_geom::geography, ' || radius || ')::geometry AS the_geom']
+              ),
+              ','
+            ) INTO selected_columns;
+            IF operation = 'create' THEN
+                -- In some cases you might want to drop the table.
```

Comment:
Totally. Probably better to remove it from the documentation. I don't have a good example for it.